test/Atmos/EDMF/edmf_model.jl

```diff
@@ -0,0 +1,345 @@
+#### EDMF model
+using DocStringExtensions
+
+#### Entrainment-Detrainment model
+
+"""
+    EntrainmentDetrainment
+
+An Entrainment-Detrainment model for EDMF, containing
+all related model and free parameters.
+
+# Fields
+$(DocStringExtensions.FIELDS)
+"""
+Base.@kwdef struct EntrainmentDetrainment{FT <: AbstractFloat}
+    "Entrainment TKE scale"
+    c_λ::FT = 0.3
+    "Entrainment factor"
+    c_ε::FT = 0.13
+    "Detrainment factor"
+    c_δ::FT = 0.52
+    "Turbulent Entrainment factor"
+    c_t::FT = 0.1
+    "Detrainment RH power"
+    β::FT = 2
+    "Logistic function scale ‵[1/s]‵"
+    μ_0::FT = 0.0004
+    "Updraft mixing fraction"
+    χ::FT = 0.25
+    "Minimum updraft velocity"
+    w_min::FT = 0.1
+    "Exponential area limiter scale"
+    lim_ϵ::FT = 0.0001
+    "Exponential area limiter amplitude"
+    lim_amp::FT = 10
+    "Minimum value for turb entr"
+    εt_min::FT = 1e-4
+end
+
+"""
+    SubdomainModel
+
+A subdomain model for EDMF, containing
+all related model and free parameters.
+
+# Fields
+$(DocStringExtensions.FIELDS)
+"""
+Base.@kwdef struct SubdomainModel{FT <: AbstractFloat}
+    "Minimum area fraction for any subdomain"
+    a_min::FT
+    "Maximum area fraction for any subdomain"
+    a_max::FT
+end
+
+function SubdomainModel(
+    ::Type{FT},
+    N_up;
+    a_min::FT = 0.001,
+    a_max::FT = 1 - a_min,
+    # a_max::FT = 1 - a_min * N_up, # TODO: switch to this?
```

Comment:
There's still a question on what the proper `a_max` is supposed to be, per [this comment](https://github.com/CliMA/ClimateMachine.jl/pull/1308#discussion_r488090782). @yairchn ?

Reply:
How do we feel about renaming `enforce_unit_bounds` to `bracket`?

Reply:
Per offline discussion. `a_max` is a valid maximum limit for _all_ subdomains, but we can/should add a maximum limit for updrafts to ensure `a_en` is not negative. Doing this is not trivial, however, so I've added some comments regarding this to revisit later.